auth-api/src/auth_api/services/user.py

```diff
@@ -63,6 +67,65 @@ def as_dict(self):
         obj = user_schema.dump(self._model, many=False)
         return obj
 
+    @staticmethod
+    def create_user_and_add_membership(memberships: List[dict], org_id, token_info: Dict = None,
+                                       skip_auth: bool = False):
+        """
+        Save a user to database and keycloak.
+
+           First add the user to keycloak
+           save the user to DB
+           create membership
+        """
+        if skip_auth:  # make sure no bulk operation and only owner is created using if no auth
+            if len(memberships) > 1 or memberships[0].get('membershipType') not in [OWNER, ADMIN]:
+                raise BusinessException(Error.INVALID_INPUT, None)
+        else:
+            check_auth(org_id=org_id, token_info=token_info, one_of_roles=(ADMIN, OWNER))
+
+        org = OrgModel.find_by_org_id(org_id)
+        if not org or org.access_type != AccessType.ANONYMOUS.value:
+            raise BusinessException(Error.INVALID_INPUT, None)
+
+        current_app.logger.debug('create_user')
+        users = []
+        for membership in memberships:
+            create_user_request = KeycloakUser()
+            username = membership['username']
+            create_user_request.user_name = username
+            create_user_request.password = membership['password']
+            try:
+                # TODO may be this method itself throw the business exception
+                kc_user = KeycloakService.add_user(create_user_request)
+            except HTTPError:
+                raise BusinessException(Error.DATA_ALREADY_EXISTS, None)
+
+            existing_user = UserModel.find_by_username(username)
+            if existing_user:
+                raise BusinessException(Error.DATA_ALREADY_EXISTS, None)
+            user_model: UserModel = UserModel()
+            user_model.username = username
+            user_model.keycloak_guid = kc_user.id
+            user_model.is_terms_of_use_accepted = False
+            user_model.status = Status.ACTIVE.value
+            user_model.type = AccessType.ANONYMOUS.value
+            user_model.email = membership.get('email', None)
+            # TODO see how to get roles
```

Comment:
Same as above.

Reply:
just dicussed this with @sumesh-aot .Roles not needed at all..will remove the todo